dask_sql/physical/rel/logical/join.py

```diff
@@ -177,8 +218,91 @@ def merge_single_partitions(lhs_partition, rhs_partition):
                     for rex in filter_condition
                 ],
             )
-            logger.debug(f"Additionally applying filter {filter_condition}")
-            df = filter_or_scalar(df, filter_condition)
+            # Three cases to deal with inequality conditions (left join as an example):
+            # Case 1 [eq_unmatched] (Not matched by equality):
+            #     Left-only from equality join (_merge=='left_only')
+            #     => Keep all
+            # Case 2 [ineq_unmatched] (Not matched by inequality):
+            #     For unique left_idx, there are no True in filter_condition
+            #     => Set values from right/left table to missing (NaN or NaT)
+            #     => Keep 1 copy and drop duplicates over left_idx (there could
+            #         be duplicates now due to equality match).
+            # Case 3 (Matched by inequality):
+            #     For unique left_idx, there are 1 or more True in filter_condition
+            #     => Keep obs with True in filter_condition
+            # This has to be added to df since partition will break the groupby
+            df["filter_condition"] = filter_condition
+            if join_type in ["left", "right"]:
+                # ----- Case 1 (Not matched by equality)
+                if join_type == "left":
+                    # Flag obs unmatched in equality join
+                    df["eq_unmatched"] = df["_merge"] == "left_only"
+                    idx_varname = "left_idx"
+                    other_varpre = "rhs_"
+                else:
+                    # Flag obs unmatched in equality join
+                    df["eq_unmatched"] = df["_merge"] == "right_only"
+                    idx_varname = "right_idx"
+                    other_varpre = "lhs_"
+
+                # ----- Case 2 (Not matched by inequality)
+
+                # Set NA (pd.NA)
+                # Flag obs not matched by inequality
+                df = df.merge(
+                    (df.groupby(idx_varname)["filter_condition"].agg("sum") < 1)
+                    .rename("ineq_unmatched")
+                    .to_frame(),
+                    left_on=idx_varname,
+                    right_index=True,
+                    how="left",
+                )
+                # Assign pd.NA
+                for v in df.columns[df.columns.str.startswith(other_varpre)]:
```

Comment:
Just tried running the tests and the same code in jupyter notebook , and got the following error ```TypeError: float() argument must be a string or a number, not 'NAType' ```, if you are not getting this error then it is probably due to environment mismatch error I guess, Or any other guess regarding this Error?
 maybe due to setting `pd.NA` ? what do you think @flcong?

Reply:
Which test triggers the error? It seems to be the conversion `float(pd.NA)`?

Reply:
For me, first query (left join) was failing in `test_join_lricomplex`
 I will try to provide more context by tomorrow :)

Reply:
OK. BTW, I use Python 3.8.10 and Pandas 1.3.2 to run the tests.

Reply:
That's a good catch @flcong, The Above Error was raised on  pandas version `1.2.4` and working fine in `pandas==1.3.2`